src/uiScreens.c

```diff
@@ -823,22 +823,22 @@ void ui_displayInputScreen(
         const sign_tx_transaction_input_t* input,
         ui_callback_fn_t callback)
 {
-	char txHex[2 * SIZEOF(input->txHashBuffer) + 1] = {0};
+	char txHex[2 * SIZEOF(input->input_data.txHashBuffer) + 1] = {0};
```

Comment:
I'm not entirely sure, but maybe we can use some temporary variables here to shorten expression like `input->input_data.txHashBuffer`.

Perhaps assert that `SIZEOF(input->input_data.txHashBuffer) == TX_HASH_LENGTH` and only use TX_HASH_LENGTH afterwards for the buffer length; and perhaps `const uint8_t* hash = input->input_data.txHashBuffer`.

Reply:
I like the TX_HASH_LENGTH idea, but for hash I think it is better to have one pointer that points to input->input_data and use it 2 times than having 2 pointers (1 for hash 1 for index) and using them once or or having pointer for one and not for the other